SingularityUI/app/components/common/formItems/FormField.jsx

```diff
@@ -4,9 +4,34 @@ import Utils from '../../../utils';
 
 let FormField = React.createClass({
 
-    render() {
-        return <input className={classNames('form-control', this.props.prop.customClass)} placeholder={this.props.prop.placeholder} type={this.props.prop.inputType} id={this.props.id} onChange={this.props.prop.updateFn} value={this.props.prop.value} disabled={this.props.prop.disabled} min={this.props.prop.min} max={this.props.prop.max} required={this.props.prop.required} />;
+  componentDidMount() {
+    if (this.props.prop.generateSelectBox) {
+      const generateSelectBox = () => {
+        $(`#${this.props.id}`).select2(this.props.prop.selectBoxOptions).on('change', this.props.prop.updateFn);
+      }
+      generateSelectBox();
+      // Delay needed because componentDidMount() is called before the entire document is rendered to HTML
+      // There is no lifecycle method that runs as soon as the document is rendered as HTML...
+      setTimeout(generateSelectBox, 1);
```

Comment:
Select2 is only a must in that it's the way the form worked in the past, and we don't want to change the styling.
@tpetr if we're going to keep select2 it could be worth it to use that wrapper, but I'd be perfectly happy to drop it if there's a better alternative.